src/noc/containers/noc-field/utils/noc-validators.ts

```diff
@@ -1,188 +1,108 @@
+        +/ 0A{2}|SAN ?TA1/.source // more postcode
+        + /$/.source
+      );
+      const isValid = value.length < REGEX_DOS_FIX_LIMIT && regEx1.test(value);
+      return isValid ? null : { postcode: true };
     };
   }
 
   public static phoneUKValidator(): ValidatorFn {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control.value === null || control.value === '') {
-        return;
-      }
-      // eslint-disable-next-line no-useless-escape
-      if (!control.value.toString().match(/^(?:(?:\(?(?:0(?:0|11)\)?[\s-]?\(?|\+)44\)?[\s-]?(?:\(?0\)?[\s-]?)?)|(?:\(?0))(?:(?:\d{5}\)?[\s-]?\d{4,5})|(?:\d{4}\)?[\s-]?(?:\d{5}|\d{3}[\s-]?\d{3}))|(?:\d{3}\)?[\s-]?\d{3}[\s-]?\d{3,4})|(?:\d{2}\)?[\s-]?\d{4}[\s-]?\d{4}))(?:[\s-]?(?:x|ext\.?|\#)\d{3,4})?$/)) {
-        return { phoneUK: true };
+      if (this.isEmpty(control.value)) {
+        return null;
       }
-      return;
+      const value = control.value.toString();
+      const regEx = new RegExp(
+        /(?:(?:\(?(?:0(?:0|11)\)?[\s-]?\(?|\+)44\)?))/.source
+      + /(([\s-]?(?:\(?0\)?[\s-]?)?)|(?:\(?0))/.source // telephone
+      + /(?:(?:\d{5}\)?[\s-]?\d{4,5})|(?:\d{4}\)))/.source
+      + /[\s-]?(?:\d{5}|\d{3}[\s-]?\d{3})/.source // optionals
+      + /(?:[\s-]?(?:x|ext\.?)\d{3,4})?$/.source
+      );
+      const isValid = value.length < REGEX_DOS_FIX_LIMIT && regEx.test(value);
+      return isValid ? null : { phoneUK: true };
     };
   }
 
   public static dateValidator(): ValidatorFn {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control.value === null || control.value === '') {
-        return;
+      if (this.isEmpty(control.value)) {
+        return null;
       }
-      const dateValues = control.value.toString().split('-');
-      if (dateValues) {
-        if (dateValues[0]) {
-          const number = Number(dateValues[0]);
-          if (number === 0 || dateValues[0].length < 4) {
-            return { date: true, year: true, valid: false };
-          }
-        } else {
-          return { date: true, year: true };
-        }
-        if (dateValues[1]) {
-          const number = Number(dateValues[1]);
-          if (number === 0 || number > 12) {
-            return { date: true, month: true, valid: false };
-          }
-        } else {
-          return { date: true, month: true };
-        }
-        if (dateValues[2]) {
-          const number = Number(dateValues[2]);
-          if (number === 0 || number > 31) {
-            return { date: true, day: true, valid: false };
-          }
-        } else {
-          return { date: true, day: true };
-        }
+      const value = control.value.toString();
+      const [year, month, day] = value.split('-').map(Number);
+      const isValidDate = moment(value, 'YYYY-MM-DD', true).isValid();
+      if (!isValidDate || year === 0 || month === 0 || month > 12 || day === 0 || day > 31) {
+        return { month: true, date: true, valid: false };
```

Comment:
The old code could return {date: true: day: true } , {date: true, year:true } or {date: true, month: true }and this code no longer does that. I think you probably need to retain this, though I'm not sure how the return value is used

Reply:
@andywilkinshmcts, the above code (date validations) works as intended when I ran some test on it locally. Please see the attached date validations images attached.

![Screenshot 2024-07-26 at 15 40 17](https://github.com/user-attachments/assets/a418c333-89ee-46d1-b547-baecdc1035aa)
![Screenshot 2024-07-26 at 15 41 00](https://github.com/user-attachments/assets/26320a12-c3d0-43f2-9d0f-5cbab7ac76bd)
![Screenshot 2024-07-26 at 15 41 24](https://github.com/user-attachments/assets/72d97808-3033-4f51-b613-337040e2d0c3)
![Screenshot 2024-07-26 at 15 41 54](https://github.com/user-attachments/assets/26c45fb2-55ad-4158-bbeb-be75218d49d6)